pkg/hvs/controllers/flavortemplate_controller_test.go

```diff
@@ -0,0 +1,434 @@
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 7,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true,
+									"eventlog_includes": [
+										"shim",
+										"db",
+										"kek",
+										"vmlinuz"
+									]
+								}
+							]
+						}
+					}
+				}`
+
+				req, err := http.NewRequest(
+					"POST",
+					"/flavor-template",
+					strings.NewReader(flavorTemplateJson),
+				)
+				req.Header.Set("Content-Type", consts.HTTPMediaTypeJson)
+				Expect(err).NotTo(HaveOccurred())
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusUnsupportedMediaType))
+			})
+		})
+
+		Context("Provide a valid FlavorTemplate data without Content-Type header", func() {
+			It("Should give HTTP Status: 415", func() {
+				router.Handle("/flavor-template", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Create))).Methods("POST")
+				flavorTemplateJson := `{
+					"label": "default-uefi",
+					"condition": [
+						"//host_info/vendor='Linux'",
+						"//host_info/tpm_version='2.0'",
+						"//host_info/uefi_enabled='true'",
+						"//host_info/suefi_enabled='true'"
+					],
+					"flavor_parts": {
+						"PLATFORM": {
+							"meta": {
+								"tpm_version": "2.0",
+								"uefi_enabled": true,
+								"vendor": "Linux"
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 0,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true,
+									"eventlog_equals": {}
+								}
+							]
+						},
+						"OS": {
+							"meta": {
+								"tpm_version": "2.0",
+								"uefi_enabled": true,
+								"vendor": "Linux"
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 7,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true,
+									"eventlog_includes": [
+										"shim",
+										"db",
+										"kek",
+										"vmlinuz"
+									]
+								}
+							]
+						}
+					}
+				}`
+
+				req, err := http.NewRequest(
+					"POST",
+					"/flavor-template",
+					strings.NewReader(flavorTemplateJson),
+				)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				Expect(err).NotTo(HaveOccurred())
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusUnsupportedMediaType))
+			})
+		})
+
+	})
+
+	// Specs for HTTP Post to "/flavor-template/{flavor-template-id}"
+	Describe("Retrieve a FlavorTemplate", func() {
+		Context("Retrieve data from valid FlavorTemplate ID", func() {
+			It("Should retrieve Flavortemplate data and get HTTP Status: 200", func() {
+				router.Handle("/flavor-template/{id}", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Retrieve))).Methods("GET")
+				req, err := http.NewRequest("GET", "/flavor-template/426912bd-39b0-4daa-ad21-0c6933230b50", nil)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
```

Comment:
Expect(err).NotTo(HaveOccurred()) must be before processing any req details

Reply:
Addressed.